### PR TITLE
Wire up setting min epoch duration

### PR DESCRIPTION
**Pull request by ashmrtn**
Add the required wiring for SDK consumers to set the min epoch duration
(and eventually other persistent config info). This is just boiler-plate
with some tests at the various layers to make sure that layer is run as
expected. More nuanced tests about setting various config values are
left to the code in `internal/kopia/conn_test.go`

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #4782

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/operations/persistent_config.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+func (op *PersistentConfigOperation) Run(ctx context.Context) (err error) {
+	defer func() {
+		if crErr := crash.Recovery(ctx, recover(), "persistent_config"); crErr != nil {
+			err = crErr
+		}
+	}()
+
+	op.Results.StartedAt = time.Now()
+
+	// TODO(ashmrtn): Send telemetry?
```
Comment (ryanfkeepers) [inline src/internal/operations/persistent_config.go:59]: what sorts of telemetry are you thinking of?

File: src/internal/kopia/conn.go [hunk 2]
```diff
@@ -38,6 +39,24 @@ const (
...
 	ErrorRepoAlreadyExists  = clues.New("repo already exists")
+
+	// minEpochDurationLowerBound is the minimum corso will allow the kopia epoch
+	// duration to be set to. This number can still be tuned further, right now
+	// it's just to make sure it's not set to something totally wild.
+	//
+	// Note that there are still other parameters kopia checks when deciding if
+	// the epoch should be changed. This is just the min amount of time since the
+	// last epoch change that's required.
+	minEpochDurationLowerBound = 2 * time.Hour
```
Comment (ryanfkeepers) [inline src/internal/kopia/conn.go:50]: question: what's an epoch / what does this configure?

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/operations/persitent_config.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+func (op *PersistentConfigOperation) Run(ctx context.Context) (err error) {
+	defer func() {
+		if crErr := crash.Recovery(ctx, recover(), "persistent_config"); crErr != nil {
+			err = crErr
+		}
+	}()
+
+	op.Results.StartedAt = time.Now()
```
Comment (pandeyabs) [inline src/internal/operations/persitent_config.go]: nit: Move start time capture to `do()` as well?

File: src/internal/operations/persitent_config.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+func (op *PersistentConfigOperation) do(ctx context.Context) error {
+	defer func() {
+		op.Results.CompletedAt = time.Now()
+	}()
+
+	err := op.operation.kopia.UpdatePersistentConfig(ctx, op.configOpts)
+	if err != nil {
+		op.Status = Failed
+		return clues.Wrap(err, "running retention config operation")
```
Comment (pandeyabs) [inline src/internal/operations/persitent_config.go]: nit: Change to `running persistent config operation`?

File: src/internal/kopia/conn.go (hunk 2 above)
Comment (ashmrtn) [reply src/internal/kopia/conn.go:50]: this is for kopia's epoch manager which holds the index mapping content IDs -> (S3 blob, blob offset). The epoch manager is kind of like a log structured merge tree, it keeps the most recent epochs "uncompacted" with lots of little files denoting the changes in the epoch (additions/removals of content). All files corresponding to an epoch need to be loaded to create a complete view of the index and access kopia's data

Once the epoch is old enough, it gets compacted into a summary of sorts. The summary is a single S3 blob that contains all the index changes for one or more epochs (kopia compacts compacted epochs to further reduce the number of S3 GETs)

This parameter denotes the minimal amount of time that has to pass before switching to a new epoch. By changing this, one can tune how many S3 blobs may need to be fetched in order to access kopia's data. This works by reducing the number of possible blobs that can accrue in an epoch prior to switching to a new epoch (and thus compacting an older one) by letting kopia switch epochs more often

File: src/internal/operations/persistent_config.go (hunk 1 above)
Comment (ashmrtn) [reply src/internal/operations/persistent_config.go:59]: oh mostly just stuff like what we have for other operation types. Not sure if we'll ever want to do something like that for this

**Review by pandeyabs — APPROVED** [review]
🚀 🚢

**Pull request merged**